Source/Core/Core/NetPlayServer.cpp

```diff
@@ -559,13 +583,46 @@ unsigned int NetPlayServer::OnData(sf::Packet& packet, Client& player)
       packet >> pad.button >> pad.analogA >> pad.analogB >> pad.stickX >> pad.stickY >>
           pad.substickX >> pad.substickY >> pad.triggerLeft >> pad.triggerRight >> pad.isConnected;
 
-      // Add to packet for relay to clients
+      if (m_host_input_authority)
+      {
+        m_last_pad_status[map] = pad;
+
+        if (!m_first_pad_status_received[map])
+        {
+          m_first_pad_status_received[map] = true;
+          SendFirstReceivedToHost(map, true);
+        }
+      }
+      else
+      {
+        spac << map << pad.button << pad.analogA << pad.analogB << pad.stickX << pad.stickY
+             << pad.substickX << pad.substickY << pad.triggerLeft << pad.triggerRight
+             << pad.isConnected;
+      }
+    }
+
+    if (!m_host_input_authority)
+      SendToClients(spac, player.pid);
+  }
+  break;
+
+  case NP_MSG_PAD_HOST_POLL:
+  {
+    sf::Packet spac;
+    spac << static_cast<MessageId>(NP_MSG_PAD_DATA);
+
+    for (PadMapping map = 0; map < 4; map++)
+    {
+      if (m_pad_map.at(map) == -1)
```

Comment:
This can just use regular indexing, no?

Reply:
Yeah I guess `at()` is only needed when we can't trust the value being passed.